api/filters/nameref/nameref.go

```diff
@@ -402,8 +402,13 @@ func (f Filter) confirmNodeMatchesReferrer(node *yaml.RNode) error {
 		return err
 	}
 	if err = checkEqual(
-		"Name", meta.Name, f.Referrer.GetName()); err != nil {
-		return err
+		"generateName",
+		meta.GenerateName,
+		f.Referrer.GetGenerateName()); err != nil {
+		if err = checkEqual(
+			"Name", meta.Name, f.Referrer.GetName()); err != nil {
+			return err
```

Comment:
Could you explain why you needed to make this change? What behavior does it affect?

Reply:
Yikes. It seemed to be a good idea at the time. All tests run just fine if you leave this change out. Originally, we had to protect GetName if there was no name, and no generateName. However,  RNode's `GetName` does the check better, to generate a suffix checksum only if GetGenerateName exists.

This needs a force push, this would look awful in git history.

Reply:
No, I was wrong (while switching back to checkEqual-ing Name only):

```
--- FAIL: TestGenerateName (0.00s)
    harness.go:100: node-referrerOriginal 'Name' mismatch '' != 'job-[0a07560513]'
--- FAIL: TestBase (0.00s)
    harness.go:100: node-referrerOriginal 'Name' mismatch '' != 'job-[320078361f]'
--- FAIL: TestMidLevelA (0.01s)
    harness.go:100: node-referrerOriginal 'Name' mismatch '' != 'job-[d66b9a897d]'
--- FAIL: TestMidLevelB (0.01s)
    harness.go:100: node-referrerOriginal 'Name' mismatch '' != 'job-[36d25ef670]'
--- FAIL: TestMultibasesNoConflict (0.01s)
    harness.go:100: node-referrerOriginal 'Name' mismatch '' != 'job-[36d25ef670]'
```

I remember now just a bit better: if `metadata.name` is empty, but `metadata.generateName` is not, `rnode.GetName()` is returning a name artificially generated (for identification purposes). In cases like this, we should better check `metadata.generateName` instead.

I enclosed calling `rnode.GetName()` with `rnode.GetGenerateName()`, which is rather simple, to save some computation time by not doing a checksum on the rendered resource for this `if` statement.